tests/test_bridge_providers.py

```diff
@@ -0,0 +1,1141 @@
+        operate = OperateApp(
+            home=tmp_path / OPERATE,
+        )
+        operate.setup()
+        operate.create_user_account(password=password)
+        operate.password = password
+        operate.wallet_manager.create(ledger_type=LedgerType.ETHEREUM)
+
+        wallet_address = operate.wallet_manager.load(LedgerType.ETHEREUM).address
+        params = {
+            "from": {
+                "chain": Chain.ETHEREUM.value,
+                "address": wallet_address,
+                "token": OLAS[Chain.ETHEREUM],
+            },
+            "to": {
+                "chain": Chain.BASE.value,
+                "address": wallet_address,
+                "token": OLAS[Chain.BASE],
+                "amount": "0",
+            },
+        }
+
+        bridge_key = "native-ethereum-to-base"
+        bridge = NativeBridgeProvider(
+            provider_id="NativeBridgeProvider",
+            bridge_contract_adaptor=OptimismContractAdaptor(
+                from_chain=NATIVE_BRIDGE_CONFIGS[bridge_key]["from_chain"],
+                from_bridge=NATIVE_BRIDGE_CONFIGS[bridge_key]["from_bridge"],
+                to_chain=NATIVE_BRIDGE_CONFIGS[bridge_key]["to_chain"],
+                to_bridge=NATIVE_BRIDGE_CONFIGS[bridge_key]["to_bridge"],
+                bridge_eta=NATIVE_BRIDGE_CONFIGS[bridge_key]["bridge_eta"],
+            ),
+            wallet_manager=operate.wallet_manager,
+        )
+
+        # Create
+        bridge_request = bridge.create_request(params)
+        expected_request = BridgeRequest(
+            params={
+                "from": {
+                    "chain": Chain.ETHEREUM.value,
+                    "address": wallet_address,
+                    "token": OLAS[Chain.ETHEREUM],
+                },
+                "to": {
+                    "chain": Chain.BASE.value,
+                    "address": wallet_address,
+                    "token": OLAS[Chain.BASE],
+                    "amount": 0,
+                },
+            },
+            bridge_provider_id=bridge.provider_id,
+            id=bridge_request.id,
+            status=BridgeRequestStatus.CREATED,
+            quote_data=None,
+            execution_data=None,
+        )
+
+        assert bridge_request == expected_request, "Wrong bridge request."
+
+        with pytest.raises(RuntimeError):
+            bridge.execute(bridge_request)
+
+        assert bridge_request == expected_request, "Wrong bridge request."
+
+        bridge._update_execution_status(bridge_request)
+
+        assert bridge_request == expected_request, "Wrong bridge request."
+
+        # Quote
+        expected_quote_data = QuoteData(
+            bridge_eta=0,
+            elapsed_time=0,
+            message=MESSAGE_QUOTE_ZERO,
+            provider_data=None,
+            timestamp=int(time.time()),
+        )
+        expected_request.quote_data = expected_quote_data
+        expected_request.status = BridgeRequestStatus.QUOTE_DONE
+
+        for _ in range(2):
```

Comment:
why twice?

Reply:
It's just an arbitrary constant meaning that I can ask quote several times without issue. On the other hand, I can't call quote after executing.